### PR TITLE
feat: Client Metrics

### PR DESCRIPTION
**Pull request by bitsandfoxes**
We could do it the same way metrics got uploaded before but then we would not dogfood the continuous upload of metrics.

### TODO
The timing needs work.

### CODE REVIEW

File: src/SymbolCollector.Android/MainActivity.cs [hunk 1]
```diff
@@ -89,10 +89,13 @@ async void OnUploadButtonOnClick(object? sender, EventArgs args)
                     uploadButton.Enabled = false;
                     source = new CancellationTokenSource();
 
-                    var uploadTask = uploader.StartUpload(_friendlyName, source.Token);
-                    var updateUiTask = StartUiUpdater(source.Token, metrics);
-
-                    await UploadAsync(uploadTask, updateUiTask, metrics, cancelButton, uploadButton, uploadTransaction, source);
+                    using (new Timing("upload-duration"))
```
Comment (bruno-garcia) [inline src/SymbolCollector.Android/MainActivity.cs:92]: Just realized the class name is `Timing` which is likely to clash?

A better API would be `using (SentrySdk.Metrics.StartTimer(".."))` cc @jamescrosswell

Comment (jamescrosswell) [reply src/SymbolCollector.Android/MainActivity.cs:92]: I thought this was solved in .NET about 20 years ago... I guess with default usings we're walking away from the use of namespaces to resolve naming conflicts. 

In any event, `SentrySdk.Metrics.StartTimer` will need to return some kind of `IDisposable` which will have to be named something. The [Python implementation uses `_Timing`](https://github.com/getsentry/sentry-python/blob/2f05ccbae7298978d2f1a8774a07386a018bcce9/sentry_sdk/metrics.py#L759). The [Sentry Developer docs](https://develop.sentry.dev/sdk/metrics/#distributions) currently say:

> For languages with context managers: `timing(key, ...)`
> For languages with lambda callback patterns: `timed(() => { ... })`
> For languages with decorators: `@timing(key, ...)` or `@timed(key, ...)`

 Do we want to rename that from `Timing` to `SentryTiming`? Then in the MetricsAggregator we could have a factory method for this called `Timed` (that would kind of match the developer docs at least)?

Comment (bruno-garcia) [reply src/SymbolCollector.Android/MainActivity.cs:92]: > In any event, SentrySdk.Metrics.StartTimer will need to return some kind of IDisposable which will have to be named something

Can be anything `SentryTimerDisposable` no one will see/interact with it

Comment (jamescrosswell) [reply src/SymbolCollector.Android/MainActivity.cs:92]: `System.Diagnostics.Metrics` doesn't have any equivalent of our `Timing` instrument:
- [Types of instruments](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/metrics-instrumentation#types-of-instruments)

In any event, I'll add an extension method on `IMetricsAggregator`:
- https://github.com/getsentry/sentry-dotnet/issues/3074

File: src/SymbolCollector.Console/ConsoleUploader.cs [hunk 2]
```diff
@@ -29,13 +29,6 @@ public async Task StartUploadSymbols(IEnumerable<string> paths, string bundleId,
         SentrySdk.ConfigureScope(s =>
         {
             s.Transaction = transaction;
-            s.AddEventProcessor(@event =>
```
Comment (bruno-garcia) [inline src/SymbolCollector.Console/ConsoleUploader.cs:32]: Why remove this though? it's helpful to see the summary at the time of the event

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**